OrbitGl/ThreadTrack.cpp

```diff
@@ -18,15 +18,41 @@ using orbit_client_protos::FunctionInfo;
 using orbit_client_protos::TimerInfo;
 
 ThreadTrack::ThreadTrack(TimeGraph* time_graph, int32_t thread_id, OrbitApp* app)
-    : TimerTrack(time_graph, app) {
+    : TimerTrack(time_graph, app), app_{app} {
```

Comment:
I'm not sure about this change. There is a protected member variable `TimerTrack::app_` which is now shadowed by `ThreadTrack::app_`. Do you really need this additional variable?

BTW, of course accessing a protected variable in a base class is very bad design, but shadowing a base class member variable is also not ideal.

Reply:
Yes. It will work just fine - as long as ThreadTrack::app_ and TimerTrack::app_ hold the same value - which is the case here. But it's suprising to us developers. There is a high risk of introducing subtle bugs in the future.

Reply:
Wait, I wouldn't shadow/overload the `app_` in ThreadTrack. From a semantical point of view, I would expect that the ` app_` in TimerTrack is the one, that this track belongs to, and that should be the same for all subclasses.

In general I see the potential issues coming with protected members, but that feels like the use-case for which protected was introduced.

Reply:
I can't completely follow. I think we might mean the same.

Reply:
I was referring to this part:
>BTW, of course accessing a protected variable in a base class is very bad design, but shadowing a base class member variable is also not ideal.